ql/src/java/org/apache/hadoop/hive/ql/parse/QBParseInfo.java

```diff
@@ -677,6 +685,33 @@ public void setNoScanAnalyzeCommand(boolean isNoScanAnalyzeCommand) {
   public boolean hasInsertTables() {
     return this.insertIntoTables.size() > 0 || this.insertOverwriteTables.size() > 0;
   }
+
+  public boolean isFullyAggregate() throws SemanticException {
```

Comment:
Although the method is evident, could we add a comment?

Should this be static since it is a utility method that can be used beyond this scope?

Reply:
Added javadocs. I think it can be non-static until it is used from another location. It uses the `destToSelExpr` field located in `QBParseInfo`.